sol/main.c

```diff
@@ -381,7 +381,9 @@ iface_speed_bytes(struct gatekeeper_if *iface, uint64_t *link_speed_bytes)
 
 	for (i = 0; i < iface->num_ports; i++) {
 		struct rte_eth_link link;
-		rte_eth_link_get(iface->ports[i], &link);
+		int ret = rte_eth_link_get(iface->ports[i], &link);
+		if (ret < 0)
+			return ret;
 
 		if (link.link_speed == ETH_SPEED_NUM_NONE) {
 			*link_speed_bytes = 0;
```

Comment:
We must set `*link_speed_bytes` to zero before returning from an error of `rte_eth_link_get()` to preserve the behavior of `iface_speed_bytes()`.

My suggestion to go about it is to add label `err:` and `goto` there for both cases. In the first case, namely the error of `rte_eth_link_get()`, we **log an error** and `goto err;`. Whereas for the second case, namely `link.link_speed == ETH_SPEED_NUM_NONE`, we set `ret = -ENOTSUP;` and `goto err;`. Notice that variable `ret` needs a bigger scope for the `goto` to work. The label `err:` will make sure that `*link_speed_bytes` is set to zero:
```c
err:
	*link_speed_bytes = 0;
	return ret;
```